bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/updater/SolarmanChannelUpdater.java

```diff
@@ -0,0 +1,246 @@
+                        + (rawVal & 0x0F))
+                .collect(Collectors.joining());
+
+        stateUpdater.updateState(channelUID, new StringType(stringValue));
+    }
+
+    private void updateChannelWithStringValue(ChannelUID channelUID, List<Integer> registers,
+            Map<Integer, byte[]> readRegistersMap) {
+        String stringValue = registers.stream().map(readRegistersMap::get).reduce(new StringBuilder(), (acc, val) -> {
+            short shortValue = ByteBuffer.wrap(val).order(ByteOrder.BIG_ENDIAN).getShort();
+            return acc.append((char) (shortValue >> 8)).append((char) (shortValue & 0xFF));
+        }, StringBuilder::append).toString();
+
+        stateUpdater.updateState(channelUID, new StringType(stringValue));
+    }
+
+    private void updateChannelWithNumericValue(ParameterItem parameterItem, ChannelUID channelUID,
+            List<Integer> registers, Map<Integer, byte[]> readRegistersMap, ValueType valueType) {
+        BigInteger value = extractNumericValue(registers, readRegistersMap, valueType);
+        BigDecimal convertedValue = convertNumericValue(value, parameterItem.getOffset(), parameterItem.getScale());
+        if (validateNumericValue(convertedValue, parameterItem.getValidation())) {
+            State state;
+
+            @Nullable
+            String uom = parameterItem.getUom();
+            if (uom == null) {
+                uom = "";
+            }
+
+            if (StringUtils.isNotEmpty(uom)) {
+                try {
+                    @Nullable
+                    Unit<?> unitFromDefinition = ChannelUtils.getUnitFromDefinition(uom);
+                    if (unitFromDefinition != null) {
+                        state = new QuantityType<>(convertedValue, unitFromDefinition);
+                    } else {
+                        logger.debug("Unable to parse unit: {}", uom);
+                        state = new DecimalType(convertedValue);
+                    }
+                } catch (MeasurementParseException e) {
+                    state = new DecimalType(convertedValue);
+                }
+            } else {
+                state = new DecimalType(convertedValue);
+            }
+            stateUpdater.updateState(channelUID, state);
+        }
+    }
+
+    private void updateChannelWithRawValue(ParameterItem parameterItem, ChannelUID channelUID, List<Integer> registers,
+            Map<Integer, byte[]> readRegistersMap) {
+        String hexString = String.format("[%s]",
+                reverse(registers).stream().map(readRegistersMap::get).map(
+                        val -> String.format("0x%02X", ByteBuffer.wrap(val).order(ByteOrder.BIG_ENDIAN).getShort()))
+                        .collect(Collectors.joining(",")));
+
+        stateUpdater.updateState(channelUID, new StringType(hexString));
+    }
+
+    private boolean validateNumericValue(@Nullable BigDecimal convertedValue, @Nullable Validation validation) {
+        return true;
+    }
```

Comment:
What is the purpose of this always returning true method?

Reply:
The original inverter definitions support validation rules (e.g., min, max) but I didn't get a chance to implement those. Removed the method and will add the validation if/when it makes sense.